contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,454 @@
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal rewards to zero to prevent reentrancy attack.
+        proposal.ethReward = 0;
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.ethReward != 0) {
+                proposal.ethRewardLeft = _proposal.ethReward;
+            }
+        }
+        amount = _proposal.ethReward;
+
+        if (amount > 0) {
+            require(Controller(avatar.owner()).sendEther(amount, _proposal.beneficiary, avatar));
+            emit RedeemEther(address(avatar), _proposalId, _proposal.beneficiary, amount);
+        }
+    }
+
+    /**
+    * @dev RedeemNativeToken reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @return amount the external token redeemed amount
+    */
+    function redeemExternalToken(bytes32 _proposalId) public returns(uint256 amount) {
+        ContributionProposal memory _proposal = organizationProposals[_proposalId];
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //set proposal rewards to zero to prevent reentrancy attack.
+        proposal.externalTokenReward = 0;
+
+        if (_proposal.beneficiary == address(this)) {
+            if (_proposal.externalTokenReward != 0) {
+                proposal.externalTokenRewardLeft = _proposal.externalTokenReward;
+            }
+        }
+
+        if (proposal.externalToken != IERC20(0) && _proposal.externalTokenReward > 0) {
+            amount = _proposal.externalTokenReward;
+            if (amount > 0) {
+                require(
+                Controller(
+                avatar.owner())
+                .externalTokenTransfer(_proposal.externalToken, _proposal.beneficiary, amount, avatar));
+                emit RedeemExternalToken(address(avatar), _proposalId, _proposal.beneficiary, amount);
+            }
+        }
+    }
+
+    /**
+    * @dev redeemReputationFromExtContract redeem reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _beneficiary the beneficiary to mint reputation to.
+    * @param _reputation the reputation amount to mint
+    */
+    function redeemReputationFromExtContract(bytes32 _proposalId, address _beneficiary, uint256 _reputation)
+    public
+    onlyRewarder
+    {
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //this will ensure sum zero of reputation.
+        proposal.reputationChangeLeft = proposal.reputationChangeLeft.sub(_reputation);
+        require(
+        Controller(
+        avatar.owner()).mintReputation(_reputation, _beneficiary, address(avatar)));
```

Comment:
Should the mint be inside of `if (_reputation != 0)`?

Should we not burn when if (_reputation < 0)?

Reply:
> Should the mint be inside of if (_reputation != 0)

we can but not must. I will add that.

> Should we not burn when if (_reputation < 0)?

the _reputation here is uint so it cannot be < 0

Reply:
See `redeemReputation`.  Why is `_reputation ` not int256 ?  (and perhaps name it without the underscore, to be consistent with `redeemReputation`)

Reply:
can we comment, like we did in `redeemReputation`, that burn is not currently allowed?